tests/test.h

```diff
@@ -49,17 +42,28 @@
 #include <sstream>
 #include <cstring>
 
-typedef boost::test_tools::predicate_result predicate_result;
+typedef bool predicate_result;
```

Comment:
I don't think this is needed any longer, i.e. it could be replaced with just `bool` in the (few, AFAICS) places where it's used.